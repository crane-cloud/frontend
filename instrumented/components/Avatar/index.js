import React from "react";
import PropTypes from "prop-types";
import "./Avatar.css";

const Avatar = ({ name,className }) => {
  const nameStringToHslColor = (name) => {
    let hash = 0;
    let i = 0;
    for (i; i < name?.length; i += 1) {
      hash = name.charCodeAt(i) + ((hash << 5) - hash); // eslint-disable-line no-bitwise
    }
    const h = hash % 360;
    return `hsl(${h}, 80%, 30%)`; // syntax: hsl(h, s%, l%)
  };

  return (
    <div
      className={className? className: "UserAvatar"}
      style={{ backgroundColor: nameStringToHslColor(name), color: "#000" }}
    >
      {name.charAt(0).toUpperCase()}
    </div>
  );
};

Avatar.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Avatar;
