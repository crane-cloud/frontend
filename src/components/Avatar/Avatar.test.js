import React from "react";
import { render } from "@testing-library/react";

import Avatar from "./index";

const sampleName = "JohnDoe";
describe("Test the avatar component", () => {
  const AvatarComponent = render(<Avatar name={sampleName} />);
  it("checks if the avatar component matches the snapshot", () => {
    expect(AvatarComponent).toMatchSnapshot();
  });
  it("checks if the avatar component correctly displays the name initial", () => {
    const UserAvatar = AvatarComponent.find("div");
    // Has class
    expect(UserAvatar.hasClass("UserAvatar")).toBe(true);
    // Renders the name initial correctly
    expect(UserAvatar.text()).toEqual(sampleName.charAt(0).toUpperCase());
  });
});
