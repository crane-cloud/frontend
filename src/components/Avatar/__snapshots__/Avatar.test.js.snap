// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the avatar component checks if the avatar component matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="UserAvatar"
        style="background-color: rgb(117, 117, 117); color: rgb(0, 0, 0);"
      >
        J
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="UserAvatar"
      style="background-color: rgb(117, 117, 117); color: rgb(0, 0, 0);"
    >
      J
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
